
# library(MSGFplus)
# library(mzID)
library(MSnbase)
library(ggplot2)
library(dplyr)
library(tibble)
library(tidyr)
library(BiocParallel)
# library(biomaRt)
library(ensembldb)
library(EnsDb.Mmusculus.v79)
library(DEqMS)
library(stringr)
library(ggtext)

####################### Parameters and Data files

centMethod<-"_msconvert"

outFolder="out/"

QvalCutoff<-0.05

paramSet="/10ppm_noPhos/"

MMdb<-EnsDb.Mmusculus.v79

exDesign<-read.table('exp-design.tsv',header=T)
rownames(exDesign)<-exDesign$reporterID

purityMx<-read.table("reporter_purity.txt", header=T, row.names=1, sep="\t") %>% as.matrix()



# MS data is deposited at ProteomeXchange under identifier PXD030748
# peptide IDs need to be generated by MS-GF+ (https://github.com/MSGFPlus/msgfplus) using
# a database of expressed retinal proteins (file refseq_expressed_retinal_proteins.fasta.gz in the current folder)
mzFiles<-list.files(path="mzML/raw/", pattern="mzML", full.names=T, recursive=F)
mzIdfiles<-list.files(paste(path="./mzML/centroided",centMethod,paramSet,sep=""), pattern=".mzid", full.names=T, recursive=F)

#######################



# Add identification data to MS3 level quantification data
rawData<-vector(mode="list", length=length(mzFiles))
for (f in 1:length(mzFiles)){ #
    print(mzFiles[f])
    idRawData<-readMSData(mzFiles[f], mode="onDisk", msLevel=3, verbose=T)
    print(mzIdfiles[f])
    mzId<-readMzIdData(mzIdfiles[f])
    protIds<-unique(mzId$DatabaseAccess)
    tmp<-genes(MMdb, filter=ProteinIdFilter(protIds))
    mC<-as.data.frame(mcols(tmp)) %>% dplyr::select(gene_id, symbol, protein_id) %>% unique()

    mzId<-left_join(mzId,mC, by=c("DatabaseAccess"="protein_id")) 
    mzId<-mzId %>% dplyr::filter(MS.GF.PepQValue<=QvalCutoff, !isDecoy, rank<=1) ### QC filter on peptides
  
    
    proteotypic<-mzId %>% arrange(acquisitionNum,MS.GF.PepQValue) %>% dplyr::select(acquisitionNum,gene_id) %>% dplyr::group_by(acquisitionNum) %>% summarise(pass=(length(unique(gene_id))==1)) 
    proteotypic<-proteotypic[proteotypic$pass,"acquisitionNum"]
    mzId<-mzId %>% dplyr::filter(acquisitionNum %in% proteotypic$acquisitionNum)
    
    mzId<-mzId %>% arrange(acquisitionNum,MS.GF.PepQValue) 
    dupIds<-mzId %>% dplyr::select(acquisitionNum) %>% duplicated()
    mzId<-mzId[!dupIds,]

    fD<-fData(idRawData)
    featureData(idRawData)<-AnnotatedDataFrame(left_join(fD,mzId,by=c("precursorScanNum"="acquisitionNum")))
    print(idSummary(idRawData))
        
    idRawData<-removeNoId(idRawData)
    idMSData<-quantify(idRawData, method="max", reporters=TMT10HCD, BPPARAM=bpparam(), verbose=T)
    idMSData<-purityCorrect(idMSData,purityMx)
    fileLabel<-paste("mzFile",f,sep="")
    idMSData<-updateFeatureNames(idMSData,fileLabel,sep=".")
    rawData[[f]]<-idMSData
}


idMSData<-rawData[[1]]
for (d in 2:length(mzFiles)){
    print(d)
    idMSData<-MSnbase::combine(idMSData,rawData[[d]])
    print(idMSData)
}

# idMSDataBackup<-idMSData

idMSData<-filterNA(idMSData,pNA=0)

################################### Combine features and normalize

#### Peptide level
idMSDataPepC<-combineFeatures(idMSData, method="sum", groupBy=fData(idMSData)$modPeptideRef,na.rm=T)
idMSDataPepCN<-normalize(idMSDataPepC,method="vsn")

## QC plot - peptide level normalization
svg(paste(outFolder,paramSet,"Peptide_level_normalization.svg",sep=""))
boxplot(exprs(idMSDataPepCN),las=2,ylab="Log2 Expression",xlab="TMT10 reporter", main="TMT10plex normalized samples, peptide level")
dev.off()

logPSM<-cbind(fData(idMSDataPepCN)[,c("sequence","gene_id","symbol","modPeptideRef")],exprs(idMSDataPepCN))
write.table(logPSM,paste(outFolder,paramSet,"logPSM_MSI_WT-KO",centMethod,".txt",sep=""),
            sep = "\t",row.names = F,quote=F)

#### Protein level
idMSDataC<-combineFeatures(idMSData, method="sum", groupBy=fData(idMSData)$gene_id,na.rm=T)
idMSDataCN<-normalize(idMSDataC,method="vsn")
exprs(idMSDataCN) %>% dim()

## QC plot - peptide level normalization
svg(paste(outFolder,paramSet,"Protein_level_normalization.svg",sep=""))
boxplot(exprs(idMSDataCN),las=2,ylab="Log2 Expression",xlab="TMT10 reporter", main="TMT10plex normalized samples, protein level")
dev.off()


logCountsTable<- cbind(fData(idMSDataCN)[,c("gene_id","symbol")],exprs(idMSDataCN))
write.table(logCountsTable,paste(outFolder,paramSet,"logCounts_MSI_WT-KO",centMethod,".txt",sep=""),
            sep = "\t",row.names = F,quote=F)
            

###################################################################################
######## DEqMS differential expression analysis ####


#### expression and phenotypic data
gene.matrix <- as.matrix(exprs(idMSDataCN))

pD<-pData(idMSData)  # idMSDataCN ?
pD$reporterID<-rownames(pD)
pD<-left_join(pD, exDesign, by="reporterID")
rownames(pD)<-pD$reporterID

##### experimental design and linear model fit
design<-model.matrix(~0+pD$Group)
contrasts<-"KO-WT"
colnames(design)<-colnames(design) %>% gsub("pD\\$Group","",.)


fit1 <- lmFit(gene.matrix,design)
contrast <- makeContrasts(contrasts=contrasts,levels=design)
fit2 <- eBayes(contrasts.fit(fit1,contrasts = contrast))

## peptide counts per gene
psmTable<- cbind(fData(idMSData)[,c("sequence","gene_id")],exprs(idMSData))
colnames(psmTable)[3:12]<-paste("X",colnames(psmTable)[3:12],sep=".")
colnames(psmTable)[1:2]<-c("Peptide","gene")
psmCountTable <- as.data.frame(table(psmTable$gene))
rownames(psmCountTable) <- psmCountTable$Var1

fit2$count <- psmCountTable[rownames(fit2$coefficients),2]
fit3 = spectraCounteBayes(fit2)
# extract DEqMS results
DEqMS.results = outputResult(fit3,coef_col = 1) 

tmp<-genes(MMdb, filter=GeneIdFilter(rownames(DEqMS.results)))
mC<-as.data.frame(mcols(tmp)) %>% dplyr::select(gene_id, symbol, entrezid) %>% unique()

DEqMS.results<-left_join(DEqMS.results,mC, by=c("gene"="gene_id")) 

head(DEqMS.results)
DEqMS.results %>% dplyr::filter(sca.adj.pval<0.05,abs(logFC)>log2(1.5)) %>% dim()
DEqMS.results%>% dim()
write.table(DEqMS.results,paste(outFolder,paramSet,"MSI_WT-KO",centMethod,".txt",sep=""),
            sep = "\t",row.names = F,quote=F)


########################### Peptide level plots
### uncoment if reading saved data
# centMethod<-"_msconvert"
exDesign<-read.table('exp-design.tsv',header=T)
logPSM<-read.table(paste(outFolder,paramSet,"logPSM_MSI_WT-KO",centMethod,".txt",sep=""),
           sep = "\t",header=T)

colnames(logPSM)[5:14]<-paste("X",colnames(logPSM)[5:14],sep=".")
psmTL2<-logPSM %>% pivot_longer(cols=starts_with("X"),names_to="reporter",values_to="Log2Intensity") 

psmTL2$reporter<-psmTL2$reporter %>% gsub("X\\.(X)*","",.)
psmTL2<-left_join(psmTL2,exDesign, by=c("reporter"="reporterID"))
# psmTL2<-left_join(psmTL2,exDesign,by=c"reporter"="reporterID")
write.table(psmTL2,paste(outFolder,paramSet,"MSI_WT-KO",centMethod,"_peptides.txt",sep=""),
            sep = "\t",row.names = F,quote=F)

       
geneSymbol="Msi1"

psmTL2 %>% dplyr::filter(symbol==geneSymbol) %>% mutate(Genotype=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=Genotype,y=Log2Intensity,color=modPeptideRef,group=rev(modPeptideRef))) +
    geom_point(stat="smooth") +
#    geom_line(stat="smooth",se=T) +
    geom_smooth (geom="line", alpha=0.2)+
    labs(color = "Peptide", y="Log2 Intensity")+
    ggtitle(paste(geneSymbol,"peptide intensities",sep=" "))+
    theme_classic()+
    theme(axis.text=element_text(size=12,face="bold"), title=element_text(size=16,face="bold"),axis.title=element_text(size=14,face="bold"), )
ggsave(paste(outFolder,paramSet,"Msi1_peptides_average.svg",sep=""),device="svg")    

psmTL2 %>% dplyr::filter(symbol==geneSymbol) %>% 
    ggplot(aes(x=reporter,y=Log2Intensity,color=modPeptideRef)) +
    geom_point() +
    geom_line(aes(group=modPeptideRef)) +
    labs(color = "Peptide",y="Log2 Intensity") +
    ggtitle(paste(geneSymbol,"peptide intensities",sep=" "))+
    theme_classic()+
    theme(axis.text=element_text(size=12,face="bold"), title=element_text(size=16,face="bold"),axis.title=element_text(size=14,face="bold"), )+
    scale_x_discrete(name="Genotype",labels=exDesign$Group)
ggsave(paste(outFolder,paramSet,"Msi1_peptides.svg",sep=""),device="svg")


geneSymbol="Msi2"

psmTL2 %>% dplyr::filter(symbol==geneSymbol) %>% mutate(Genotype=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=Genotype,y=Log2Intensity,color=modPeptideRef,group=rev(modPeptideRef))) +
    geom_point(stat="smooth") +
#    geom_line(stat="smooth",se=T) +
    geom_smooth (geom="line", alpha=0.2)+
    labs(color = "Peptide", y="Log2 Intensity") +
    ggtitle(paste(geneSymbol,"peptide intensities",sep=" "))+
    theme_classic()+
    theme(axis.text=element_text(size=12,face="bold"), title=element_text(size=16,face="bold"),axis.title=element_text(size=14,face="bold"), )
ggsave(paste(outFolder,paramSet,"Msi2_peptides_average.svg",sep=""),device="svg")    

 
psmTL2 %>% dplyr::filter(symbol==geneSymbol) %>% 
    ggplot(aes(x=reporter,y=Log2Intensity,color=modPeptideRef)) +
    geom_point() +
    geom_line(aes(group=modPeptideRef)) +
    labs(color = "Peptide",y="Log2 Intensity") +
    ggtitle(paste(geneSymbol,"peptide intensities",sep=" "))+
    theme_classic()+
    theme(axis.text=element_text(size=12,face="bold"), title=element_text(size=16,face="bold"),axis.title=element_text(size=14,face="bold"), )+
    scale_x_discrete(name="Genotype",labels=exDesign$Group)
ggsave(paste(outFolder,paramSet,"Msi2_peptides.svg",sep=""),device="svg")


####### Protein expression boxplots
### uncoment if reading saved data
# centMethod<-"_msconvert"
#exDesign<-read.table('exp-design.tsv',header=T)
logCountsTable<-read.table(paste(outFolder,paramSet,"logCounts_MSI_WT-KO",centMethod,".txt",sep=""),
           sep = "\t",header=T)

## scale to controls
normFact<- logCountsTable[,3:7] %>% rowMeans() 
logCountsTable[,3:12]<-logCountsTable[,3:12]-normFact
write.table(logCountsTable,paste(outFolder,paramSet,"MSI_WT-KO_normalized_protein_expr_",centMethod,".txt",sep=""),
            sep = "\t",row.names = F,quote=F)
            
colnames(logCountsTable)[4:13]<-paste("X",colnames(logCountsTable)[4:13],sep=".")


normCounts<-logCountsTable %>% pivot_longer(cols=starts_with("X"),names_to="reporter",values_to="Log2FC") 
normCounts$reporter<-normCounts$reporter %>% gsub("X\\.|(X)*","",.)
normCounts<-left_join(normCounts,exDesign[,c("reporterID","AnimalID","Genotype","Group")], by=c("reporter"="reporterID"))
normCounts$AnimalID<-as.factor(normCounts$AnimalID)
normCounts$Group<-factor(normCounts$Group,levels=c("WT","KO"))
normCounts$symbol<-normCounts$symbol %>% gsub("Gm11744","Prcd",.)

genes=c("Msi1","Msi2")
normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("Protein Level Change")+
    facet_wrap(vars(symbol),ncol=3,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,1)
ggsave("Musashi_protein_level_change.svg",device="svg")


genes=c("Gnat1", "Cnga1","Guca1b","Rcvrn")
normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>% 
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("Phototransduction Proteins")+
    facet_wrap(vars(symbol),ncol=4,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,2)
ggsave("Phototransduction_protein_level_change.svg",device="svg")


genes=c("Rom1","Prcd","Prom1","Prph2")
normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("Outer Segment Proteins")+
    facet_wrap(vars(symbol),ncol=4,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,2)
ggsave("OS_protein_level_change.svg",device="svg")


genes=c("Cc2d2a","Cep290","Rpgr","Mak")
normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("Cilia Proteins")+
    facet_wrap(vars(symbol),ncol=4,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,2)
ggsave("Cilia_protein_level_change.svg",device="svg")


genes=c("Ift80","Ift122","Ift172","Ift43")
    normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("IFT Proteins")+
    facet_wrap(vars(symbol),ncol=4,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,2)
ggsave("IFT_protein_level_change.svg",device="svg")

genes=c("Ttc8","Bbs2","Bbs1","Bbs9")
    normCounts %>% dplyr::filter(symbol %in% genes) %>% mutate(`Genotype `=str_replace_all(Genotype,",","\n")) %>%
    ggplot(aes(x=`Genotype `,y=Log2FC)) +
    geom_boxplot(aes(color=Genotype),outlier.shape = NA)+
    geom_jitter()+
    theme_classic() +
    ggtitle("BBSome Proteins")+
    facet_wrap(vars(symbol),ncol=4,scales="fixed")+
    labs(y=expression(bold(paste(Log[2], " Fold Change")))) +
    theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
    ylim(-2,2)
ggsave("BBSome_protein_level_change.svg",device="svg")



#### MA plot
DEqMS.results$sig="Unchanged"
DEqMS.results$sig[DEqMS.results$sca.adj.pval<=0.05 & DEqMS.results$logFC<(-log2(1.5))]="Downregulated"
DEqMS.results$sig[DEqMS.results$sca.adj.pval<=0.05 & DEqMS.results$logFC>log2(1.5)]="Upregulated"

DEqMS.results %>% 
    ggplot(aes(x=AveExpr,y=logFC,color=sig)) +
    geom_point() +
    scale_fill_manual(breaks=c("Downregulated","Unchanged","Upregulated"), values=c("Red","Gray","Blue"))    +
    scale_color_manual(breaks=c("Downregulated","Unchanged","Upregulated"), values=c("Red","Darkgray","Blue")) +
    theme_classic()
ggsave("MA-plot-proteomics.svg",device="svg")
    

    
############# Combined RNA-Seq/Proteomics analysis

#load RNASeq data
excludedGenes<-c("Msi1","Msi2","Esr1","Pde6g") ## knockout targets Msi1 and Msi2; Cre-ERT component Esr1; Pde6g alele is hemizygous (Cre-ERT is knockin)

colnames(DEqMS.results)<-gsub("$",".MS",colnames(DEqMS.results))

RNAExpr<-read.csv('../Floxed_vs_Msi1-2KO_Differential_expression.csv',header=T,row.names=1)
RNAExpr$ensemble_gene_id<-rownames(RNAExpr)
colnames(RNAExpr)<-gsub("$",".RNASeq",colnames(RNAExpr))
# sigRNA<-RNAExpr %>% dplyr::filter(FDR<=0.05, abs(logFC)>=log2(1.5)) #%>% rownames()


MsRna<-left_join(DEqMS.results, RNAExpr, by=c("gene.MS"="ensemble_gene_id.RNASeq"))

MsRna$sig.MS<-(MsRna$sca.adj.pval.MS<=0.05 & abs(MsRna$logFC.MS)>=log2(1.5))
MsRna$sig.RNASeq<-(MsRna$FDR.RNASeq<=0.05 & abs(MsRna$logFC.RNASeq)>=log2(1.5))

MsRna$`Expression Change`<-factor(NA,levels=c("Not Significant","Protein","RNA","RNA and Protein"))

MsRna$`Expression Change`<-levels(MsRna$`Expression Change`)[(1+MsRna$sig.MS*1 + MsRna$sig.RNASeq*2)]

MsRna %>% dplyr::filter(!(symbol.MS %in% excludedGenes)) %>%
    ggplot(aes(x=logFC.RNASeq, y=logFC.MS, color=`Expression Change`)) +
    geom_point() +
    theme_classic() +
    scale_color_manual(values=c("Darkgray","Blue","Green","Orange"))



##### Genes with high levels of expression specifically in photoreceptors from scRNASeq
#########################################################################
#### Convert the supplementary table from Macosko 2015

fName<-'retina_clusters_Macosko.txt'
fHandle<-file('retina_clusters_Macosko.txt',"r")
fSize<-system2("wc", args=c("-l",fName), stdout=T) %>% strsplit(.," ") %>% unlist() %>% as.integer()
fSize=fSize[1]

geneClusters<-data.frame(matrix(NA,nrow=fSize,ncol=4))
colnames(geneClusters)<-c("gene","myP","myDif","Clusters")
rowNum=1
while ( TRUE ) {
    if ( length(line) == 0 ) {
      break
    }
    line <- readLines(fHandle, n = 1)
    fields<-strsplit(line,"\t") %>% unlist()
#     if (all(fields=="")) { next}
    if (fields[1]=="Clusters") { 
        cluster=fields[2]
        next
        }
    if (all(fields!="")){
        fields[2:3]<-as.numeric(fields[2:3])
        geneClusters[rowNum,c("gene","myP","myDif","Clusters")]<-c(fields,cluster)
        rowNum=rowNum+1
    }
    if( (rowNum %% 1000)==0){ print(c(rowNum, 100*rowNum/fSize))}
}
close(fHandle)

geneClusters<-geneClusters[complete.cases(geneClusters),]

geneClusters<-geneClusters %>% mutate(gene=ifelse(!(length(gene)>=10 & str_ends(gene,"RIK")),str_to_title(gene),gene))
write.table(geneClusters,'flat_retina_clusters_Macosko.txt',row.names=F, sep='\t')



##########################################################################################
### identify the photoreceptor specific genes. Clusters 24 and 25 are rods and cones respectively
geneClusters<-read.table('flat_retina_clusters_Macosko.txt',header=T, sep="\t")

geneClusters<-geneClusters %>% separate(Clusters, into=c("ClusterA","ClusterB"),sep="\\+",convert=T)

## Photoreceptor genes
prClusters<-geneClusters %>% dplyr::filter((ClusterA %in% c(24,25)) | (ClusterB %in% c(24,25)) & !(ClusterA %in% c(24,25)) & (ClusterB %in% c(24,25)))
gcA<-prClusters %>% dplyr::filter(ClusterA %in% c(24,25))
gcB<-prClusters %>% dplyr::filter(ClusterB %in% c(24,25))
gcB<-gcB %>% mutate(myDif=(-myDif)) 
colnames(gcB)[c(4,5)]<-colnames(gcB)[c(5,4)]

prClusters<-bind_rows(gcA,gcB)
prGenes<-prClusters %>% dplyr::group_by(gene) %>% summarise(`Photoreceptor protein`=all(myDif>0)) 


## Other genes
excludedClusters<-c(24,25) 
otherGenes<-geneClusters %>% dplyr::group_by(gene) %>% summarise(`Other protein`=(!any(ClusterA %in% excludedClusters) & !any(ClusterB %in% excludedClusters))) 


MsRna<-left_join(MsRna,prGenes,by=c("symbol.MS"="gene"))
MsRna<-left_join(MsRna,otherGenes,by=c("symbol.MS"="gene"))

MsRna %>% dplyr::filter(!(symbol.MS %in% excludedGenes), !is.na(`Photoreceptor protein`), !is.na(`Other protein`)) %>%
    ggplot(aes(x=logFC.RNASeq, y=logFC.MS, color=`Expression Change`)) +
    geom_point(aes(shape=`Photoreceptor protein`,stroke=1)) +
    theme_classic() +
    scale_color_manual(values=c("Darkgray","Blue","Green","Orange"))+
    scale_shape_manual(values=c(3,5))+
    xlim(-2.1,2.1) + ylim(-2.1,2.1)
 ggsave("RNA_vs_Protein_levels.svg",device="svg")   




################### cumulative frequency plots


#### Setup for cumulative frequency plots

logFC.MS<-MsRna %>% dplyr::filter(`Photoreceptor protein`) %>% pull(logFC.MS)
logFC.RNA<-MsRna %>% dplyr::filter(`Photoreceptor protein`) %>% pull(logFC.RNASeq)
ksTestPR<-ks.test(logFC.MS,logFC.RNA,alternative="greater")

logFC.MSother<-MsRna %>% dplyr::filter(`Other protein`) %>% pull(logFC.MS)
ksTestOther<-ks.test(logFC.MS,logFC.MSother,alternative="greater")


prCumSumProt<-MsRna %>% dplyr::filter(`Photoreceptor protein`) %>% dplyr::group_by(logFC.MS)  %>% summarise(n=n(),logFC=logFC.MS,Expression="Protein Photoreceptor") %>% mutate(n=cumsum(n)/sum(n)) %>% ungroup() %>% dplyr::select(-logFC.MS)
prCumSumRNA<-MsRna %>% dplyr::filter(`Photoreceptor protein`) %>% dplyr::group_by(logFC.RNASeq) %>% summarise(n=n(),logFC=logFC.RNASeq,Expression="RNA Photoreceptor") %>% mutate(n=cumsum(n)/sum(n)) %>% ungroup() %>% dplyr::select(-logFC.RNASeq)


otherCumSumProt<-MsRna %>% dplyr::filter(`Other protein`) %>% dplyr::group_by(logFC.MS)  %>% summarise(n=n(),logFC=logFC.MS,Expression="Protein Other") %>% mutate(n=cumsum(n)/sum(n)) %>% ungroup() %>% dplyr::select(-logFC.MS)
otherCumSumRNA<-MsRna %>% dplyr::filter(`Other protein`) %>% dplyr::group_by(logFC.RNASeq) %>% summarise(n=n(),logFC=logFC.RNASeq,Expression="RNA Other") %>% mutate(n=cumsum(n)/sum(n)) %>% ungroup() %>% dplyr::select(-logFC.RNASeq)


cumSum<-rbind(prCumSumProt,otherCumSumProt,prCumSumRNA,otherCumSumRNA)


pvalPR=formatC(ksTestPR$p.value, format="e", digits=2) %>% str_replace(.,"e(.*)","<sup>\\1</sup>")
cumSum %>% dplyr::filter(str_detect(Expression,"Photoreceptor")) %>% 
    ggplot( aes(x=logFC, y=n,  color=Expression)) + 
        geom_line(size=2) + 
        ylab("Cumulative Frequency") + 
        xlab("logFC") + 
        xlim(-1.5,1.5)+
        theme_light()+
        theme(legend.title=element_blank())+
        theme(text=element_text(size=14),
        axis.text=element_text(size=12,face="bold"), 
        title=element_text(size=16,face="bold"),
        axis.title=element_text(size=14,face="bold"),
        strip.text.x=element_text(size=14,face="bold"))+
        annotate(geom="richtext",x=(-1.5), y=0.95, label=paste0("<b><span style = 'font-size:14pt'>Kolmogorov-Smirnov p-value ",pvalPR,"</span></b>"),hjust=0)
ggsave("PR_protein_RNA_expression.svg",device="svg")




